relentless/potential/pair.py

```diff
@@ -561,3 +561,230 @@ def _derivative(self, param, r, epsilon, kappa, **params):
+
+        if s:
+            u = u.item()
+        return u
+
+    def _force(self, r, P, sigma_i, sigma_j, sigma_d, **params):
+        """Evaluates the depletion force.
+
+        Parameters
+        ----------
+        r : array_like
+            The value or values of r at which to evaluate the force.
+        P : float or int
+            The P parameter for the potential function.
+        sigma_i : float or int
+            The sigma_i parameter for the potential function.
+        sigma_j : float or int
+            The sigma_j parameter for the potential function.
+        sigma_d : float or int
+            The sigma_d parameter for the potential function.
+
+        Returns
+        -------
+        array_like
+            Returns the value or values of the force evaluated at r.
+
+        Raises
+        ------
+        ValueError
+            If sigma_i, sigma_j, and sigma_d are not all positive.
+
+        """
+        if sigma_i<=0 or sigma_j<=0 or sigma_d<=0:
+            raise ValueError('sigma_i, sigma_j, and sigma_d must all be positive')
+        r,f,s = self._zeros(r)
+        lo = 0.5*(sigma_i + sigma_j)
+        hi = lo + sigma_d
+
+        #clamp lo
+        lo = 0.5*(sigma_i + sigma_j)
+        f[np.less(r,lo)] = (-np.pi*P*sigma_i*sigma_j*sigma_d*(sigma_i + sigma_j + sigma_d)
+                           /(sigma_i + sigma_j)**2)
+
+        #clamp hi
+        hi = lo + sigma_d
+        f[np.greater(r,hi)] = 0
+
+        # evaluate in between
+        flags = np.logical_and(np.greater_equal(r,lo), np.less_equal(r,hi))
+        p1 = r[flags]**2 - 0.25*(sigma_i - sigma_j)**2
+        p2 = (0.5*(sigma_i + sigma_j) + sigma_d)**2 - r[flags]**2
+        f[flags] = -(np.pi*P*p1*p2)/(4*r[flags]**2)
+
+        if s:
+            f = f.item()
+        return f
+
+    def _derivative(self, param, r, P, sigma_i, sigma_j, sigma_d, **params):
+        """Evaluates the depletion parameter derivative.
+
+        Parameters
+        ----------
+        param : `str`
+            The parameter with respect to which to take the derivative.
+        r : array_like
+            The value or values of r at which to evaluate the derivative.
+        P : float or int
+            The P parameter for the potential function.
+        sigma_i : float or int
+            The sigma_i parameter for the potential function.
+        sigma_j : float or int
+            The sigma_j parameter for the potential function.
+        sigma_d : float or int
+            The sigma_d parameter for the potential function.
+
+        Returns
+        -------
+        array_like
+            Returns the value or values of the derivative evaluated at r.
+
+        Raises
+        ------
+        ValueError
+            If sigma_i, sigma_j, and sigma_d are not all positive.
+        ValueError
+            If the parameter with respect to which to take the derivative
+            is not P, sigma_i, sigma_j, or sigma_d.
+
+        """
```

Comment:
Same suggestions as for `_energy` apply here. Obviously the functions are a bit more complicated. :-)